src/main/java/org/folio/circulation/domain/notice/schedule/DueDateNotRealTimeScheduledNoticeHandler.java

```diff
@@ -109,20 +118,33 @@ public DueDateNotRealTimeScheduledNoticeHandler(
       .map(Pair::getRight)
       .map(LoanAndRelatedRecords::getLoan)
       .map(TemplateContextUtil::createLoanNoticeContextWithoutUser)
-      .collect(Collectors.toList());
+      .collect(toList());
 
     //All the notices have the same properties so we can get any of them
     ScheduledNotice scheduledNotice = relevantNotices.get(0).getLeft();
     LoanAndRelatedRecords noticeRelatedRecords = relevantNotices.get(0).getRight();
+    Loan loan = noticeRelatedRecords.getLoan();
 
     User user = noticeRelatedRecords.getLoan().getUser();
     JsonObject noticeContext = new JsonObject()
       .put("user", TemplateContextUtil.createUserContext(user))
       .put("loans", new JsonArray(loanContexts));
 
-    return patronNoticeService.acceptScheduledNoticeEvent(
-      scheduledNotice.getConfiguration(), user.getId(), noticeContext)
-      .thenApply(mapResult(v -> noticeGroup));
+    NoticeLogContext noticeLogContext = NoticeLogContext.from(loan)
+      .withItems(relevantNotices.stream()
+        .map(Pair::getRight)
+        .map(LoanAndRelatedRecords::getLoan)
+        .map(NoticeLogContextItem::from)
+        .collect(toList()))
+      .withTriggeringEvent(scheduledNotice.getTriggeringEvent().getRepresentation())
+      .withTemplateId(scheduledNotice.getConfiguration().getTemplateId());
+
+    return noticePolicyRepository.lookupPolicyId(loan.getItem(), loan.getUser())
```

Comment:
So, this method handles group of notices, but we only retrieve notice policy for the first element of the group. The policy can be different depending on user and item. I guess, we should retrieve the policy for each notice somewhere before this method, can it be stored in the `ScheduledNotice` object?

Reply:
Yes, it would be better to store it in the `ScheduledNotice`